project/src/test/java/com/google/sps/VolunteerOpportunityTest.java

```diff
@@ -0,0 +1,79 @@
+package com.google.sps;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.sps.data.VolunteeringOpportunity;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
+@RunWith(JUnit4.class)
+public final class VolunteerOpportunityTest {
+  private static final String NAME = "Performer";
+  private static final int NUMBER_OF_SPOTS = 240;
+  private static final String PLAYING_AN_INSTRUMENT = "Playing an instrument";
+  private static final String PERFORMING = "Performing";
+  private static final String DANCING = "Dancing";
+  private static final Set<String> SKILLS = ImmutableSet.of(PLAYING_AN_INSTRUMENT, PERFORMING);
+  private static final Set<String> SKILLS_WITH_DANCING =
+      ImmutableSet.of(PLAYING_AN_INSTRUMENT, PERFORMING, DANCING);
+
+  @Test
+  public void createInstanceUsingBuilder() {
+    VolunteeringOpportunity opportunity =
+        new VolunteeringOpportunity.Builder(NAME, NUMBER_OF_SPOTS)
+            .setRequiredSkills(SKILLS)
+            .build();
+
+    Assert.assertEquals(NAME, opportunity.getName());
+    Assert.assertEquals(NUMBER_OF_SPOTS, opportunity.getNumSpotsLeft());
+    Assert.assertEquals(SKILLS, opportunity.getRequiredSkills());
+  }
+
+  @Test
+  public void getBuilderFromInstance() {
+    VolunteeringOpportunity opportunity =
+        new VolunteeringOpportunity.Builder(NAME, NUMBER_OF_SPOTS)
+            .setRequiredSkills(SKILLS)
+            .build();
+    VolunteeringOpportunity copyOfOpportunity = opportunity.toBuilder().build();
+
+    Assert.assertEquals(NAME, copyOfOpportunity.getName());
+    Assert.assertEquals(NUMBER_OF_SPOTS, copyOfOpportunity.getNumSpotsLeft());
+    Assert.assertEquals(SKILLS, copyOfOpportunity.getRequiredSkills());
+  }
+
+  @Test
+  public void addSkillsWithBuild() {
+    Set<String> skills = new HashSet<String>();
+    skills.add(PLAYING_AN_INSTRUMENT);
+    skills.add(PERFORMING);
+    VolunteeringOpportunity opportunityWithSkills =
+        new VolunteeringOpportunity.Builder(NAME, NUMBER_OF_SPOTS)
+            .setRequiredSkills(skills)
+            .build();
+    VolunteeringOpportunity changedOpportunity =
+        opportunityWithSkills.toBuilder().addRequiredSkill(DANCING).build();
+
+    Assert.assertEquals(SKILLS_WITH_DANCING, changedOpportunity.getRequiredSkills());
```

Comment:
Could you maybe find something test contains?

Looks like:
assertThat(changedOpportunity.getRequiredSkills(), containsString(PLAYING_AN_INSTRUMENT, PERFORMING, DANCING));

Reply:
Done - had to add hamcreast dependency for assertThat.